### DISCUSSION TITLE
Evaluator Rules - condition to evaluate only direct and one level below package dependencies

### DISCUSSION THREAD

**Discussion by felix-leyva** [oss-review-toolkit/ort#6724] [Q&A]
Hello ORT team,
I am writing an evaluator.rules.kts file and would like to ask you, if there is any way, to only evaluate within a packageRule, dependencies which are directly declared in a Gradle Project, or which are the first level dependencies of a dependency of a project.
Ex:
Root Project
  - Package 1
    - Package1.1
      - Package1.1.1
      - Package1.1.2
    - Package1.2
      -  ...
    - ...
  - Package 2
    - Package 2.1
    - ...
The following should be evaluated: Package1, 1.1, 1.2, 2, 2.1
All other below should not evaluated.

Thanks for your help, best regards
Felix

**Comment by sschuberth** [comment]
> to only evaluate within a packageRule, dependencies

Not that this is sort of a contradiction. If you want to analyze dependencies (i.e. references to packages), you should use a [DependencyRule](https://github.com/oss-review-toolkit/ort/blob/781882079a14e484bcdb73b10432bdd41055e7a8/evaluator/src/main/kotlin/DependencyRule.kt#L30-L33) instead of a [PackageRule](https://github.com/oss-review-toolkit/ort/blob/781882079a14e484bcdb73b10432bdd41055e7a8/evaluator/src/main/kotlin/PackageRule.kt#L35-L38). 

The former has

https://github.com/oss-review-toolkit/ort/blob/781882079a14e484bcdb73b10432bdd41055e7a8/evaluator/src/main/kotlin/DependencyRule.kt#L73-L76

which should suite your needs.
